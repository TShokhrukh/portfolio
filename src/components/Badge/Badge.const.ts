import { cn } from '@bem-react/classname'

export const cnBadge = cn('Badge')
