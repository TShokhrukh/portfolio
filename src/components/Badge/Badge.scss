
.Badge {
  padding: 5px 10px;
  border-radius: $border-radius;
  font-weight: bold;
}