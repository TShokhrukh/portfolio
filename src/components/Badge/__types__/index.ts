import type { ReactNode, ElementType } from 'react'
import type { IClassNameProps } from '@bem-react/core'

export interface IBadgeProps extends IClassNameProps {
  role?: string,
  id?: string,
  component?: ElementType,
  children?: ReactNode
}

export interface IBadgeColorProps {
  color?: 'purple'
}
