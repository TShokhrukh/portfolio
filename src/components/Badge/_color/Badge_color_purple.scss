
.Badge_color_purple {
  background-color: rgba($purple, 0.1);
  color: $purple;
}