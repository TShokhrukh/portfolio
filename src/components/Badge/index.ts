export * from './build'
export * from './__types__'
export { Badge as BadgeBase } from './Badge'
