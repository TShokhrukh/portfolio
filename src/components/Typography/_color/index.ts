export * from './Typography_color_dark'
export * from './Typography_color_purple'
export * from './Typography_color_rose'
export * from './Typography_color_white'
