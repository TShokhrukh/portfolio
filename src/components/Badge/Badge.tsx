import React from 'react'
import { cnBadge } from './Badge.const'
import type { IBadgeProps } from './__types__'

import './Badge.scss'

export const Badge: React.FC<IBadgeProps> = ({
  component: Component = 'span',
  className,
  children,
  ...props
}) => {
  return (
    <Component {...props} className={cnBadge({}, [className])}
      >{children}</Component>
  )
}
