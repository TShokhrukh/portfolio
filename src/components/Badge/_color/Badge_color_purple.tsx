import React from 'react'
import { withBemMod } from '@bem-react/core'
import { cnBadge } from '../Badge.const'

import type { IBadgeColorProps, IBadgeProps } from '../__types__'

import './Badge_color_purple.scss'

export const withBadgeColorPurple = withBemMod<IBadgeColorProps, IBadgeProps>(
  cnBadge(),
  { color: 'purple' },
  // eslint-disable-next-line react/prop-types
  (Badge) => function BadgeColorPurple ({ color: _color, ...props }) {
    return <Badge {...props} />
  }
)
