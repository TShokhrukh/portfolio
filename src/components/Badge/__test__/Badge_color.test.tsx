/* global test, expect, describe */
import React from 'react'
import { compose } from '@bem-react/core'
import { render, screen } from '@testing-library/react'
import { Badge as BadgeBase } from '../Badge'
import { withBadgeColorPurple } from '../_color/Badge_color_purple'

const Badge = compose(withBadgeColorPurple)(BadgeBase)

describe('Badge_color', () => {
  test('component template', () => {
    render(<Badge role='badge' />)

    expect(screen.getByRole('badge')).toMatchSnapshot()
  })

  test('component color purple', () => {
    render(<Badge data-testid='badge' color='purple' />)

    expect(screen.getByTestId('badge')).toHaveAttribute('class', 'Badge Badge_color_purple')
  })
})
