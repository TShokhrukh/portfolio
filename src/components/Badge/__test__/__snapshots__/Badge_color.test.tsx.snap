// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge_color component template 1`] = `
<span
  class="Badge"
  role="badge"
/>
`;
